src/pmhn/_trees/_tree_utils.py

```diff
@@ -0,0 +1,189 @@
+from anytree import Node, RenderTree, LevelOrderGroupIter
+from itertools import combinations, product
+from typing import Optional
+
+
+def all_combinations_of_elements(*lists):
+    """
+    Takes a variable number of lists as input and returns a list
```

Comment:
I read the docstring twice and I'm still unsure what the method does (and where it is used).

Perhaps rephrasing the docstring or adding a small example there:
```
For example, for [1, 2], [10, 11, 12] this function will generate...
```
would help in the long run.